hack/generator/pkg/astmodel/file_definition_test.go

```diff
@@ -28,7 +28,7 @@ func NewTestStruct(name string, fields ...string) StructDefinition {
 	}
 
 	ref := NewStructReference(name, "group", "2020-01-01", false)
-	definition := NewStructDefinition(ref, fs...)
+	definition := NewStructDefinition(ref, NewStructType(fs...))
```

Comment:
Why does `NewStructType()` need to be called here vs embedding that detail within `NewStructDefinition()`?

Reply:
Because a struct type now has references to functions as well, and you can't have 2 varargs in a function I don't think, and even if we could, I don't think we'd want to.

The fact the definition has-a StructType isn't really a surprise as it's clear from the public definition, what's the goal of hiding it in the new?

Reply:
That makes sense. Not sure I've got an alternative, but it looks a bit clunky to me.